open-sphere-base/mantle/src/main/java/io/opensphere/mantle/plugin/selection/SelectionCommandFactory.java

```diff
@@ -0,0 +1,266 @@
+    {
+        if (DEFAULT_COMMAND_MAP.containsKey(name))
+        {
+            return DEFAULT_COMMAND_MAP.get(name);
+        }
+//        else if (CLASSNAME_COMMAND_MAP.containsKey(name))
+//        {
+//            return CLASSNAME_COMMAND_MAP.get(name);
+//        }
+        // Unknown command returned.
+        LOGGER.warn("Illegal Selection Command Recieved: " + name);
+        return null;
+    }
+
+    /**
+     * Gets the no geometry menu items.
+     *
+     * @param al the {@link ActionListener}
+     * @return the no geometry menu items
+     */
+    public static List<Component> getNoGeometryMenuItems(ActionListener al)
+    {
+        List<Component> menuItems = new ArrayList<>();
+        menuItems.add(DESELECT.createMenuItem(al));
+        menuItems.add(CREATE_BUFFER_REGION.createMenuItem(al));
+        return menuItems;
+    }
+
+    /**
+     * Gets the point menu items.
+     *
+     * @param al the {@link ActionListener}
+     * @param hasFilters Whether there are filters associated with the menu.
+     * @return the point menu items
+     */
+    public static List<Component> getPointMenuItems(ActionListener al, boolean hasFilters)
+    {
+        List<Component> menuItems = new ArrayList<>();
+        menuItems.add(CREATE_BUFFER_REGION.createMenuItem(al));
+        return menuItems;
+    }
+
+    /**
+     * Gets the selection region menu items.
+     *
+     * @param listener A listener interested in actions on the menu items.
+     * @param hasFilters Whether there are filters associated with the menu.
+     * @param isFromCreateBuffer true when the menu should include the create
+     *            buffer option.
+     * @return the selection region menu items
+     */
+    public static List<Component> getPolygonMenuItems(ActionListener listener, boolean hasFilters, boolean isFromCreateBuffer)
+    {
+        List<Component> menuItems = new ArrayList<>();
+        menuItems.add(createHeader("QUERY"));
+        menuItems.add(ADD_FEATURES.createMenuItem(listener, hasFilters ? FILTERS_ACTIVE : null));
+        menuItems.add(LOAD_FEATURES.createMenuItem(listener, hasFilters ? FILTERS_ACTIVE : null));
+        menuItems.add(createHeader("FEATURES"));
+        menuItems.add(DESELECT.createMenuItem(listener));
+        menuItems.add(REMOVE_ALL.createMenuItem(listener));
+        menuItems.add(SELECT.createMenuItem(listener));
+        menuItems.add(SELECT_EXCLUSIVE.createMenuItem(listener));
+        if (!isFromCreateBuffer)
+        {
+            menuItems.add(createHeader("TOOLS"));
+            menuItems.add(CREATE_BUFFER_REGION.createMenuItem(listener));
+        }
+        return menuItems;
+    }
+
+    /**
+     * Gets the polyline menu items.
+     *
+     * @param al the {@link ActionListener}
+     * @param hasFilters Whether there are filters associated with the menu.
+     * @return the polyline menu items
+     */
+    public static List<Component> getPolylineMenuItems(ActionListener al, boolean hasFilters)
```

Comment:
This function also doesn't use hasFilters.

Reply:
fixed